build_tools/i18n/crowdin.py

```diff
@@ -283,36 +284,48 @@ def _csv_to_json():
         csv_locale_dir_path = os.path.join(
             utils.local_locale_csv_path(), lang_object["crowdin_code"]
         )
-        for file_name in os.listdir(csv_locale_dir_path):
-            if file_name.endswith("json"):
-                # Then it is a Perseus JSON file - just copy it.
-                source = os.path.join(csv_locale_dir_path, file_name)
-                target = os.path.join(perseus_path, file_name)
-                try:
-                    os.makedirs(perseus_path)
-                except:
-                    pass
-                shutil.copyfile(source, target)
-                continue
-            elif not file_name.endswith("csv"):
+        perseus_locale_dir_path = os.path.join(
+            utils.local_perseus_locale_csv_path(), lang_object["crowdin_code"]
+        )
+
+        # Make sure that the Perseus directory for CSV_FILES/{lang_code} exists
+        if not os.path.exists(perseus_locale_dir_path):
+            os.makedirs(perseus_locale_dir_path)
+
+        csv_dirs = os.listdir(csv_locale_dir_path) + os.listdir(perseus_locale_dir_path)
+
+        for file_name in csv_dirs:
+            if "csv" not in file_name:
                 continue
 
-            csv_path = os.path.join(csv_locale_dir_path, file_name)
+            if file_name is PERSEUS_CSV:
+                csv_path = os.path.join(perseus_locale_dir_path, file_name)
+            else:
+                csv_path = os.path.join(csv_locale_dir_path, file_name)
 
             # Account for csv reading differences in Pythons 2 and 3
-            if sys.version_info[0] < 3:
-                csv_file = open(csv_path, "rb")
-            else:
-                csv_file = open(csv_path, "r", newline="")
+            try:
+                if sys.version_info[0] < 3:
+                    csv_file = open(csv_path, "rb")
+                else:
+                    csv_file = open(csv_path, "r", newline="")
```

Comment:
Could this be addressed somehow with `io.open` instead of the conditional?

Not sure if it's related, but in general using `io.open` has had better python2/3 compatibility: https://github.com/learningequality/kolibri/pull/3769

Reply:
No, the newline doesn't get handled by that - it's due to a change in the csv library between 2 and 3.

Reply:
okay good to know.

That said, perhaps this should still be written using `io.open`?

```python
newline = None if sys.version_info[0] < 3 else ""
csv_file = io.open(csv_path, mode="r", encoding="utf-8", newline=newline)
```

Reply:
Hrm, I know a guy who prefers the more verbose option usually... https://github.com/learningequality/kolibri/pull/5990#discussion_r339157314

Reply:
But yeah, `io.open` is preferable.

Reply:
This is how I did it for some csv work elsewhere: https://github.com/learningequality/kolibri/blob/develop/kolibri/core/auth/csv_utils.py#L196

Reply:
> Hrm, I know a guy who prefers the more verbose option usually... #5990 (comment)

hahha not in this case! That construction was specifically to avoid the [repeated evaluation](https://github.com/learningequality/kolibri/pull/5990#discussion_r339162355) problem you mentioned, which is not the case here

Reply:
Also needs the `b` conditionally too, so I think it is still neater with the if else.

Reply:
> Also needs the `b` conditionally too

Is this something specific to the CSV library? My understanding was that specifying `encoding="utf-8"` made the `b` unnecessary